types/index-test.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint-disable node/no-missing-import */
+
+import OktaSDK from 'OktaSDK';
+
+// $ExpectType Promise<boolean>
+OktaSDK.createConfig({
```

Comment:
How does this file work? Is this for validating in an editor?

Reply:
I introduced this file (and tool) to test a declaration file (`.d.ts`). The file is used by `dtslint` tool. Somehow we must assert that declaration file works and meets our expectation regarding parameters, naming, returned types.

Also, it checks the declaration file for [some rules](https://github.com/microsoft/dtslint/tree/master/docs) to keep it clean.